packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts

```diff
@@ -126,35 +110,27 @@ export default class AttributeSlicer extends VisualBase {
     private isHandlingUpdate: boolean;
 
     /**
-     * Constructor
+     * The previous update options
      */
-    constructor(noCss = false, options: any) {
-        super("Attribute Slicer", noCss);
-        this.state = VisualState.create() as VisualState;
-
-        options.element = $(options.element); // make this a jquery object
-        this.init(options);
-    }
+    private prevUpdateOptions: powerbi.extensibility.visual.VisualUpdateOptions;
 
     /**
-     * Gets the template associated with the visual
+     * Constructor
      */
-    public get template() {
-        return `<div></div>`;
-    }
+    constructor(options: powerbi.extensibility.visual.VisualConstructorOptions, noCss = false) {
+        this.state = VisualState.create() as VisualState;
+        this.host = options.host;
+        this.element = $("<div></div>");
 
-    /**
-     * Called when the visual is being initialized
-     */
-    public init(options: powerbi.VisualInitOptions): void {
-        super.init(options);
+        // Add to the container
+        options.element.appendChild(this.element[0]);
 
-        this.host = options.host;
-        this.selectionManager = this.host["createSelectionManager"]();
+        this.selectionManager = this.host.createSelectionManager();
         this.propertyPersister = createPropertyPersister(this.host, 100);
 
         const className = CUSTOM_CSS_MODULE && CUSTOM_CSS_MODULE.locals && CUSTOM_CSS_MODULE.locals.className;
         if (className) {
+            $(options.element).append($("<st" + "yle>" + CUSTOM_CSS_MODULE + "</st" + "yle>"));
```

Comment:
why do we have "<st" + "yle>" ?

Reply:
At some point, there was some PBI code or something that caused the stylesheets not to load without that trick, not sure if it is necessary anymore.